### ISSUE TITLE
I ran into some problems when using VSCODE to knit RMD

### ISSUE THREAD

**Issue by Sgrcaivnemn** [rstudio/bookdown#1461]
Here is my problem:
<img width="342" alt="b6bc65fa47860e4bb657350b9cbd495" src="https://github.com/rstudio/bookdown/assets/137772681/04330951-5c85-4f23-bd06-83a1a6bd3392">

**Comment by cderv** [comment]
Thanks for the report. 

Unfortunately, with only this screenshot, we won't be able to do much. This is indicates that you indeed encountered an issue, but it could be anything. 

You should at least look in the "R Markdown output stream" - it should be accessible in one of the panes of VS CODE (Output pane with the selector then on the right). This will give you the message. 

The error could come from **bookdown**, **rmarkdown**, **knitr** or just one of your code cell execution, or any of your Rmd content. 

So to sum up, we need more information in order for us to look into this. Please help us help you. Thank you !

**Comment by svix424** [comment]
i'm having the same problem and get the following error: 

Error: pandoc version 1.12.3 or higher is required and was not found (see the help page ?rmarkdown::pandoc_available).

Execution halted

[VSC-R] HW 8.Rmd process exited with exit code 1
[VSC-R] terminating R process

any suggestions welcome!

**Comment by cderv** [comment]
> Error: pandoc version 1.12.3 or higher is required and was not found (see the help page ?rmarkdown::pandoc_available).

Do you have Pandoc installed and available on your system PATH ? 

This message tells you that required version is not found - did you check for it ?

**Comment by svix424** [comment]
I followed [these instructions](https://stackoverflow.com/questions/74018894/error-pandoc-version-1-12-3-or-higher-is-required-in-vscode-but-works-in-r-s) and set a new system PATH to this: RSTUDIO_PANDOC = "C:/Program Files/RStudio/bin/quarto/bin/tools/". 
It creates reprex.rmd files, and a .txt file that still gives me the same message: Error : pandoc version 1.12.3 or higher is required and was not found (see the help page ?rmarkdown::pandoc_available).

**Comment by cderv** [comment]
> set a new system PATH to this: RSTUDIO_PANDOC = "C:/Program Files/RStudio/bin/quarto/bin/tools/".

What does this means ? Did you add a new environment variable ? Or added the bin path to the PATH environment variable ? 

`RSTUDIO_PANDOC` env var should be used by R Markdown - you can check with `rmarkdown::find_pandoc()`
What does it return ? 

Also did you restart VSCODE or your computer to be sure the env var change taken into account ? 

If you have only pandoc available in RStudio through pandoc, instead of pointing to that version you could either 

- Install pandoc on your system and put in PATH env var
- Install quarto on your system and put the bin/tools in env var
- Using **pandoc** R package to manage your pandoc installation, and configure R Markdown to use it (or set in PATH env var)

**Comment by katrinabrock** [comment]
I use vscode and here is the snippet that I run for each session where I need to use rmarkdown (after one-time installing pandoc with the pandoc R package as mentioned above):

```{r}
pandoc::pandoc_activate(); Sys.setenv("RSTUDIO_PANDOC"=rmarkdown::find_pandoc()$dir)
```

I'm using Linux, but should be platform agnostic. You could put this in your `.Rprofile` for example.

**Comment by katrinabrock** [comment]
Not sure why it's not getting picked up.

Weirdly now I can't repro...but I've definitely hit this issue in the past. If/when I hit it again, I'll debug it a bit further.

**Comment by khoidnutsw** [comment]
Hello,

I'm encountering the same issue with RMarkdown previews in VS Code (running on a Linux HPC environment via Remote-SSH). When I try to "Open Preview" on an .Rmd file, the "R Markdown" output log shows the following error:

[VSC-R] debug.Rmd process started
Error: pandoc version 1.12.3 or higher is required and was not found (see the help page ?rmarkdown::pandoc_available).
Execution halted
However, I can successfully render the same .Rmd file to HTML (or other formats) via the command line using R CMD rmarkdown::render() with same conda environment. The pandoc path is verified.

I would greatly appreciate any suggestions on how to address this.
Thank you

**Comment by cderv** [comment]
> When I try to "Open Preview" on an .Rmd file, the "R Markdown" output log shows the following error:

This seems to be something related to VSCODE extension which has support for bookdown and R Markdown. 

This type of error usually means that Pandoc is not found by `rmarkdown::render()` in the VSCODE context. You should look into that. 

> However, I can successfully render the same .Rmd file to HTML (or other formats) via the command line using R CMD rmarkdown::render() with same conda environment. The pandoc path is verified.

The fact that you can render via command line means the R package works as expected. 

You could also ask toward the developers and community of the Extension in question.
